src/views/Pathways/Add/Course.js

```diff
@@ -0,0 +1,77 @@
+import React, {useEffect} from 'react';
+import { useSnackbar } from 'notistack';
+import { withRouter } from 'react-router';
+import { bindActionCreators } from "redux";
+import { connect } from "react-redux";
+import { compose } from 'recompose';
+import {find} from 'lodash';
+import { addOrEditCourse } from '../store';
+import { selectors as layoutSelectors, setRightPaneLoading } from '../../../layouts/TwoColumn/store';
+import RightPaneWithTitle from '../../../components/RightPaneWithTitle';
+import FormBuilder from '../../../components/FormBuilder';
+import history from '../../../providers/routing/app-history';
+import { getCourseAddFormStructure } from '../forms/course';
+import { ngFetch } from '../../../providers/NGFetch';
+
+
+const CourseAdd = ({ actions, match }) => {
+  const {pathwayId} = match.params;
+  const { enqueueSnackbar } = useSnackbar();
+
+  const [CoursesAlreadyPresent, setCoursesAlreadyPresent] = React.useState(null);
+  useEffect(() => {
+    const fetchallCourses = async () => {
+      actions.setRightPaneLoading(true);
+      const response = await ngFetch(`/pathways/${pathwayId}/courses`);
+      setCoursesAlreadyPresent(response);
+      actions.setRightPaneLoading(false);
+    }
+    fetchallCourses();
+  }, [actions,pathwayId]);
+
+  console.log(CoursesAlreadyPresent,'CoursesAlreadyPresent')
+
+  const [submitBtnDisabled, setSubmitBtnDisabled] = React.useState(false);
+
+  const [allCourses, setallCourses] = React.useState(null);
+  useEffect(() => {
```

Comment:
The work happening in two `useEffects` here can effectively happen in only one `useEffect`. Also the courses that need to be displayed in the `select` can be computed directly in `useEffect` and then given to the form. This will make things easier.